### PR TITLE
docs: troubleshooting

### PR DESCRIPTION
**Pull request by phoracek**
**What this PR does / why we need it**:

This is the last article from the introduction series. It describes the negative flow, how to debug a failed configuration.

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by oshoval** [comment]
best seller soon xD

File: docs/user-guide-103-troubleshooting.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+spec:
+  desiredState:
+  interfaces:
+  - name: eth666
+    type: ethernet
+    state: up
+```
+
+```shell
+kubectl apply -f eth666_up.yaml
```
Comment (oshoval) [inline docs/user-guide-103-troubleshooting.md]: O_o

File: test/e2e/demos_test.go [hunk 2]
```diff
@@ -23,26 +23,51 @@ func skipIfNotKubernetes() {
 	}
 }
 
-var _ = Describe("Introduction: Configuration", func() {
+var _ = Describe("Introduction", func() {
```
Comment (qinqon) [inline test/e2e/demos_test.go]: Maybe it's super stupid idea, but would be nice to have a test that fails if docs has being updated but test are not.

**Comment by qinqon** [comment]
@phoracek please rebase

File: test/e2e/demos_test.go (hunk 2 above)
Comment (phoracek) [reply test/e2e/demos_test.go]: The tests are using the same examples as docs. And docs have a note saying "update the example if you update the .md". I hope that is enough, we just need to keep in mind to update tests. I don't consider it worthy an advanced mechanism to check diffs in doc. KISS

**Comment by phoracek** [comment]
/override pull-kubernetes-nmstate-e2e-k8s

Not related, due to the previous doc test. Has to be fixed separately.

**Pull request merged**